### PR TITLE
68355182 - [WIP] event news letter feature

### PR DESCRIPTION
**Pull request by StlMaris123**
Link to the story: https://www.pivotaltracker.com/story/show/68355182

### CODE REVIEW

**Comment by tansaku** [comment]
great work getting started on this @StlMaris123 :-)

File: features/step_definitions/events_newsletter_steps.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+When(/^"([^"]*)" opens the email$/) do | user|
+  user = User.where(superadmin: false)
+end
+
+Then(/^I should see email delivered from "([^"]*)"$/) do |sender|
+  expect(email).to have_sender(sender)
```
Comment (tansaku) [inline features/step_definitions/events_newsletter_steps.rb]: so email here will have to de defined, or be an instance variable or passed in as an argument

**Comment by tansaku** [comment]
@StlMaris123 this is great progress.  Is it failing in the way you expect?

**Comment by StlMaris123** [comment]
@tansaku the test should be failing since they have not yet been implemented

File: features/step_definitions/events_newsletter_steps.rb (hunk 1 above)
Comment (StlMaris123) [reply features/step_definitions/events_newsletter_steps.rb]: okay will look into that

**Comment by tansaku** [comment]
@StlMaris123 yes it should fail, but at the moment it's failing due to an internal coding error, not because the functionality is absent - do you know what I mean?

**Comment by tansaku** [comment]
great work @StlMaris123 - I have it working on my side.  I can see the email going out in the test.log:

```
EventNewsLetterMailer#monthly_newsletter: processed outbound mail in 171.1ms

Sent mail to nonsuperadmin@myorg.com (10.8ms)
Date: Mon, 04 Dec 2017 16:34:47 +0000
From: support@harrowcn.org.uk
Reply-To: support@harrowcn.org.uk
To: nonsuperadmin@myorg.com
Cc: technical@harrowcn.org.uk
Message-ID: <5a25792794666_14aa3fd36183fa08135f0@Samuels-MBP.home.mail>
Subject: Upcoming events updates
Mime-Version: 1.0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<html>
  <body>
    
  </body>
</html>
```

I guess the thing we need to consider is whether we are going to be sending things out via the rails mailer, or if we are going to be using the MailChimp API ...

**Comment by StlMaris123** [comment]
This is great. We will discuss it in the scrum

**Comment by tansaku** [comment]
hi @StlMaris123 - any plans to start working on this again? or should we give to someone else?

**Comment by StlMaris123** [comment]
Hi Sam, Someone else could do it but please tag me a long. I would love to
see how it has been implemented

**Comment by nisevi** [comment]
Hi there, as discussed with @StlMaris123 I created a fresh start here https://github.com/AgileVentures/LocalSupport/pull/598

**Pull request closed without merging**